src/lbp.jl

```diff
@@ -0,0 +1,146 @@
+function lbp_original(bit_pattern::Array{Bool, 1})
+	sum([b * 2 ^ (length(bit_pattern) - i) for (i, b) in enumerate(bit_pattern)])
+end
+
+UNIFORM_LBP_TABLE = Dict{Array{Bool, 1}, Int}()
+UNIFORM_PATTERN_COUNT = 0
+NON_UNIFORM_PATTERN = Array{Bool, 1}()
+
+function init_uniform_lbp_params(points::Integer)
+	global UNIFORM_LBP_TABLE, NON_UNIFORM_PATTERN, UNIFORM_PATTERN_COUNT
+	
+	UNIFORM_PATTERN_COUNT::Int = 0
+	empty!(UNIFORM_LBP_TABLE)
+	NON_UNIFORM_PATTERN = zeros(Bool, points)
+	NON_UNIFORM_PATTERN[1:2:points] = true
+	UNIFORM_LBP_TABLE[NON_UNIFORM_PATTERN] = points * (points - 1) + 2
+end
+
+function lbp_uniform(bit_pattern::Array{Bool, 1})
+	global UNIFORM_LBP_TABLE, NON_UNIFORM_PATTERN, UNIFORM_PATTERN_COUNT
+	variations = sum([bit_pattern[i] != bit_pattern[i + 1] for i in 1:length(bit_pattern) - 1])
+	if variations <= 2
+		try 
+			return UNIFORM_LBP_TABLE[bit_pattern] 
+		catch KeyError 
+			UNIFORM_PATTERN_COUNT::Int += 1
+			UNIFORM_LBP_TABLE[bit_pattern] = UNIFORM_PATTERN_COUNT::Int
+			return UNIFORM_PATTERN_COUNT::Int 
+		end
+	else
+		return UNIFORM_LBP_TABLE[NON_UNIFORM_PATTERN]	
+	end
+end
+
+function lbp_rotation_invariant(bit_pattern::Array{Bool, 1})
+	mini = lbp_original(bit_pattern)
+	for i in 2:length(bit_pattern)
+   	   mini = min(mini, lbp_original(vcat(bit_pattern[i:end], bit_pattern[1:i-1])))
+    end
+    mini
+end
+
+function _lbp{T<:Gray}(img::AbstractArray{T, 2}, points::Integer, offsets::Array, method::Function = lbp_original)
+	init_uniform_lbp_params(points)
+	lbp_image = zeros(UInt, size(img))
+	R = CartesianRange(size(img))
+	for I in R
+		bit_pattern = [img[I] < bilinear_interpolation(img, I[1] + o[1], I[2] + o[2]) ? false : true for o in offsets]
```

Comment:
You'd probably be much better off pre-allocating the `bit_pattern` array and then storing the results in it. One issue, though, is that when you use it as a key for `UNIFORM_LBP_TABLE` you'll have to make a copy (see above).